azkaban-common/src/main/java/azkaban/imagemgmt/daos/RampRuleDaoImpl.java

```diff
@@ -0,0 +1,105 @@
+package azkaban.imagemgmt.daos;
+
+import azkaban.db.DatabaseOperator;
+import azkaban.imagemgmt.exception.ErrorCode;
+import azkaban.imagemgmt.exception.ImageMgmtDaoException;
+import azkaban.imagemgmt.exception.ImageMgmtException;
+import azkaban.imagemgmt.exception.ImageMgmtInvalidInputException;
+import azkaban.imagemgmt.models.ImageRampRule;
+import azkaban.imagemgmt.models.ImageType;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.apache.log4j.Logger;
+import org.bouncycastle.util.Strings;
+
+
+@Singleton
+public class RampRuleDaoImpl implements RampRuleDao {
+  private static final Logger LOG = Logger.getLogger(RampRuleDaoImpl.class);
+  private final DatabaseOperator databaseOperator;
+
+  private static String INSERT_RAMP_RULE = "INSERT into ramp_rules "
+      + "(rule_id, image_name, image_version, owners, is_HP, created_by, created_on) values (?, ?, ?, ?, ?, ?, ?)";
+
+  @Inject
+  public RampRuleDaoImpl(DatabaseOperator databaseOperator) {
+    this.databaseOperator = databaseOperator;
+  }
+
+  @Override
+  public boolean isHPRule(String ruleId) {
+    return false;
+  }
+
+  @Override
+  public int addRampRule(ImageRampRule rampRule) {
+    try {
+      List<ImageRampRule> rampRules = databaseOperator.query(
```

Comment:
I don't think we need to manually check the duplicate in java level. Since you defined rule_id as PRIMARY KEY, it is by default UNIQUE at db level and thus if duplicate, db will throw the error back directly.

Reply:
I agree here. We should handle the error and return to user if insert fails.

Reply:
I tested this will be returned as any other SQLException and, I think internal SQL error would be marked as 500 server error, while this duplicated request would be categorized as client error, and falls in to 400 BadRequest.

Reply:
Sounds good to me